src/tools/pkcs15-tool.c

```diff
@@ -1961,8 +1961,11 @@ static int test_update(sc_card_t *in_card)
 			}
 			/* other tag */
 			i += 2 + rbuf[2+i+1]; /* length of this tag*/
+			if (2+i+1 >= sizeof(rbuf)) {
+				break;
+			}
 		}
-		if (rbuf[2+i+1] < 9 || 2+i+2+9 > apdu.resplen) {
+		if (2+i+1 >= sizeof(rbuf) || rbuf[2+i+1] < 9 || 2+i+2+9 > apdu.resplen) {
```

Comment:
Having another look, I think the comparison here as well as above should be done against the `apdu.resplen` instead of `sizeof(rbuf)` to capture also the cases where the fuzzer returns less data than the size of `rbuf` and the rest of the `rbuf` stays uninitialized.

Reply:
Indeed, fixed